### PR TITLE
app/vmalert: do not wait for group start on removal

### PR DESCRIPTION
**Pull request by hagen1778**
Each group in vmalert starts with an artifical delay to avoid
thundering herd problem. For some groups with high evaluation
intervals, the delay could be significant.
If during this delay user will remove the group from the config
and hot-reload it - vmalert will have to wait until the delay
ends. This results into slow config reloading and UI hang.

The change moves the start-delay logic back to the group's
`start` method. Now, group can immediately exit from the
delay when `group.close()` method is called.

### CODE REVIEW

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

File: app/vmalert/manager.go [hunk 1]
```diff
@@ -100,9 +100,11 @@ func (m *manager) startGroup(ctx context.Context, g *Group, restore bool) error
 			select {
 			case <-ctx.Done():
 				sleepTimer.Stop()
+				close(g.finishedCh)
 				return
 			case <-g.doneCh:
 				sleepTimer.Stop()
+				close(g.finishedCh)
```
Comment (f41gh7) [inline app/vmalert/manager.go:107]: Could it panic? Looks like channel could be closed twice:

```
start(ctx context.Context, nts func() []notifier.Notifier, rw *remotewrite.Client, rr datasource.QuerierBuilder) {
	defer func() { close(g.finishedCh) }()
```

Comment (hagen1778) [reply app/vmalert/manager.go:107]: I agree it looks not rly cool, but there will be no panic.
This chan can be closed either here or inside of `group.start()` func which is called inside of this routine. So the execution here is sequential.
Alternative approach would be only maintaining this chan inside this one function, but then `group.start()` will be harder to use in tests.

Comment (hagen1778) [reply app/vmalert/manager.go]: @f41gh7 please see PR updated

**Review by f41gh7 — APPROVED** [review]
LGTM

**Pull request merged**